File: extensions/offlineAds/src/main/java/com/mxplay/offlineads/exo/oma/Ad.java

package com.mxplay.offlineads.exo.oma;

import androidx.annotation.Nullable;

public interface Ad {
    String getAdId();
    @Nullable
    String getCreativeId();
    boolean isSkippable();
    double getSkipTimeOffset();
    String getDescription();
    String getTitle();
    @Nullable String getAdvertiserName();
    long getDuration();
    AdPodInfo getAdPodInfo();
    String getMediaUrl();
}

File: extensions/mxInteractiveMediaAds/src/main/java/com/mxplay/interactivemedia/api/Ad.kt
package com.mxplay.interactivemedia.api

import com.mxplay.interactivemedia.api.player.AdMediaInfo
import com.mxplay.interactivemedia.internal.tracking.ITrackersProvider

interface Ad : ITrackersProvider {
    fun getAdId(): String?
    fun getSequence(): Int
    fun getCreativeId(): String?
    fun isSkippable(): Boolean
    fun getSkipTimeOffset(): Long
    fun getDescription(): String?
    fun getTitle(): String?
    fun getAdvertiserName(): String?
    fun getDuration(): Long
    fun getAdPodInfo(): AdPodInfo
    fun getMediaInfo(): AdMediaInfo?
    fun getCompanionAds(): List<CompanionAd>?
    fun getVastMediaWidth(): Int
    fun getVastMediaHeight(): Int
    fun getTraffickingParameters(): String?
}